Source/SourceKittenFramework/Loader.swift

```diff
@@ -0,0 +1,145 @@
+//
+//  Loader.swift
+//  sourcekitten
+//
+//  Created by Norio Nomura on 2/20/16.
+//  Copyright © 2016 SourceKitten. All rights reserved.
+//
+
+import Foundation
+
+struct DynamicLinkLibrary {
+    let path: String
+    let handle: UnsafeMutablePointer<Void>
+    
+    func loadSymbol<T>(symbol: String) -> T {
+        let sym = Darwin.dlsym(handle, symbol)
+        if sym == nil {
+            let errorString = String(UTF8String: dlerror())
+            fatalError("Finding symbol \(symbol) failed: \(errorString)")
+        }
+        return unsafeBitCast(sym, T.self)
+    }
+}
+
+let toolchainLoader = Loader(paths: [
+    xcodeDefaultToolchainOverride,
+    toolchainDir,
+    xcodeSelectPath?.toolchainDir,
+    /*
+    Followings are used when `xcode-select -p` points "Command Line Tools OS X for Xcode",
+    but Xcode.app exists.
+    */
+    applicationsDir?.xcodeDeveloperDir.toolchainDir,
+    applicationsDir?.xcodeBetaDeveloperDir.toolchainDir,
+    userApplicationsDir?.xcodeDeveloperDir.toolchainDir,
+    userApplicationsDir?.xcodeBetaDeveloperDir.toolchainDir,
+    ].flatMap { path in
+        if let fullPath = path?.usrLibDir where fullPath.isFile {
+            return fullPath
+        }
+        return nil
+    })
+
+struct Loader {
+    let paths: [String]
+
+    func load(path: String) -> DynamicLinkLibrary {
+        let fullPaths = paths.map { $0.stringByAppendingPathComponent(path) }.filter { $0.isFile }
+
+        // try all fullPaths that contains target file
+        for fullPath in fullPaths {
+            let handle = dlopen(fullPath, RTLD_LAZY)
+            if handle != nil {
+                return DynamicLinkLibrary(path: path, handle: handle)
+            }
+        }
+
+        // try loading with simple path that depends resolving to DYLD
+        let handle = dlopen(path, RTLD_LAZY)
+        if handle != nil {
+            return DynamicLinkLibrary(path: path, handle: handle)
+        }
+
+        fatalError("Loading \(path) failed")
+    }
+}
+
+/// Returns "XCODE_DEFAULT_TOOLCHAIN_OVERRIDE" environment variable
+///
+/// `launch-with-toolchain` set toolchain path to "XCODE_DEFAULT_TOOLCHAIN_OVERRIDE" environment
+/// variable.
+private let xcodeDefaultToolchainOverride: String? =
+    NSProcessInfo.processInfo().environment["XCODE_DEFAULT_TOOLCHAIN_OVERRIDE"]
```

Comment:
Note: I'v found that Xcode 7.3 beta 4 (7D152p) does not have `launch-with-toolchain`.
```sh
% DEVELOPER_DIR="/Applications/Xcode-beta.app" xcrun launch-with-toolchain
2016-02-23 15:35:53.988 xcodebuild[95151:1105215] [MT] PluginLoading: Required plug-in compatibility UUID E71C2CFE-BFD8-4044-8F06-00AE685A406C for plug-in at path '~/Library/Application Support/Developer/Shared/Xcode/Plug-ins/ExtraBuildPhase.xcplugin' not present in DVTPlugInCompatibilityUUIDs
xcrun: error: unable to find utility "launch-with-toolchain", not a developer tool or in PATH
```